### PR TITLE
Remove elements in multi-valued attributes - SCIM2

### PR DESCRIPTION
**Pull request by AnuradhaSK**
## Purpose
Remove elements in multi-valued attributes - SCIM2
Fix for wso2/product-is#8135
Public fix for https://github.com/wso2-support/charon/pull/45/

## Related PRs
Depends on https://github.com/wso2/carbon-kernel/pull/2705/

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java [hunk 1]
```diff
@@ -245,10 +245,46 @@ private static AbstractSCIMObject doPatchRemoveWithFiltersForLevelThree(Abstract
...
+                    // This is only valid for extensions.
+                    Attribute subSubAttribute = subAttribute.getSubAttribute(attributeParts[2]);
+                    if (subSubAttribute == null) {
+                        String error =
+                                String.format("No such sub attribute with the name : %s within the attribute %s",
+                                        attributeParts[2], attributeParts[1]);
+                        throw new BadRequestException(error, ResponseCodeConstants.INVALID_PATH);
+                    }
+                    if (!subSubAttribute.getMultiValued()) {
+                        String error = String.format("Attribute : %s is not a multivalued attribute.",
```
Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java:257]: We dont need to use String.format if we are only appending a single element

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java [hunk 2]
```diff
@@ -245,10 +245,46 @@ private static AbstractSCIMObject doPatchRemoveWithFiltersForLevelThree(Abstract
...
+                                attributeParts[2]);
+                        throw new BadRequestException(error, ResponseCodeConstants.INVALID_PATH);
+                    }
+
+                    // We only support "EQ" filter.
+                    if (!(SCIMConstants.OperationalConstants.VALUE).equals(expressionNode.getAttributeValue()) ||
+                            !((SCIMConstants.OperationalConstants.EQ).trim())
```
Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java:264]: Why do we need to trim a constant?

File: modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java [hunk 3]
```diff
@@ -131,6 +131,17 @@ public User updateUser(User user, Map<String, Boolean> map)
        }
     }
 
+    public User updateUser(User user, Map<String, Boolean> requiredAttributes,
+                           List<String> allSimpleMultiValuedAttributes)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        if (user.getId() == null) {
+            throw new NotFoundException("No user found with the id : " + user.getId());
```
Comment (somindatommy) [inline modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java:139]: The value here is null right? If so why we try to print the value? I think it should say Empty userID. Also, can we use StringUtils?

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java (hunk 2 above)
Comment (AnuradhaSK) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java:264]: The values of these constants contain leading and trailing spaces for some other usage. eg:  `public static final String EQ = " eq ";`

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java (hunk 1 above)
Comment (AnuradhaSK) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java]: addressed in https://github.com/wso2/charon/pull/286/commits/f738971766186c1ccf09b23ccb0cee39cacd016a

File: modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java (hunk 3 above)
Comment (AnuradhaSK) [reply modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java]: addressed in https://github.com/wso2/charon/pull/286/commits/7def504e293f80abeda86855908072c643671d50

**Review by somindatommy — APPROVED** [review]

**Comment by somindatommy** [comment]
@AnuradhaSK Add the PR builder job as a comment

**Comment by AnuradhaSK** [comment]
Charon is packed into product-is by scim component. Thus PR builder doesn't make the build successful. 
Locally all integration tests got passed.

**Pull request merged**